WIP: Initial glossary from workshop.

happi opened the pull request:
WIP
Please review and comment.

If you have some excellent diagrams to make things clearer, feel free to add them.

When satisfied, we should copy this glossary to the aeternity repo and check the code to ensure we use the correct terminology.

happi committed:
Initial glossary from workshop.

ThomasArts commented on glossary.md:
```diff
@@ -0,0 +1,66 @@
+Contains one keyblock and possibly one microblock
+
+### Keyblock
+A placeholder block indicating the leader that should produce microblocks in this generation.
```
"generation" and "plural micro blocks is confusing here.
```suggestion
A placeholder block indicating the leader that should produce a microblock.
```

ThomasArts commented on glossary.md:
```diff
@@ -0,0 +1,66 @@
+A block with transactions.
+
+### Epoch
+Period of time represented in blocks during which validators are pre-selected to produce blocks and after which the state of the CC can be pinned to the PC.
```
Possibly call this a "CC Epoch then".
There's also epoch for PC, but that differs in definition

happi committed:
Update glossary.md

Co-authored-by: Thomas Arts <thomas.arts@quviq.com>

ThomasArts commented on glossary.md:
```diff
@@ -0,0 +1,66 @@
+Election rules defined at contract level (as opposed to consensus level). Likely not part of v1.0.
+
+### Genesis Accounts
+Accounts in which HC tokens are generated during the first block of CC.
```
In our case it also contains some contracts used in the CC... or actually, no, but we have the notion of genesis contracts or genesis block as well

happi replied:
Could you formulate that as a change request?

happi committed:
Addressing some comments.

happi committed:
Add cycle diagram.

happi commented:
The diagram looks nice in my VSCode, but it is a bit messed up on GitHub. Do you get the right view, or should I add a rendered picture?
![image](https://github.com/user-attachments/assets/d8916ab0-27b3-4e30-aa99-ecc5a1afbc15)

happi committed:
Fix cycle diagram.

happi committed:
Headings.

happi committed:
Move block production.

thoughtfulcool committed:
Update glossary.md

Added Initiator description.

thoughtfulcool committed:
Update glossary.md

Added Hyperchains description.

Edited and augmented Child Chain and Parent Chain definitions for clarity.

dincho commented on glossary.md:
```diff
@@ -0,0 +1,133 @@
+### Pinning Reward
+Tokens minted and/or transferred to staking contract to reward successful pinning operation.
+### Halting
+A situation when no more blocks are produced (due to all stakers being inactive). This creates a problem for the HC creator to recover the HC and resume the block production. Some possible solutions are proposed.
```
Isn't stakers defined as passive actors above? Shouldn't this refer to validators?

> due to all validators being inactive

hanssv commented on glossary.md:
```diff
@@ -0,0 +1,133 @@
+
+
+### Leader Election Contract
+Election rules defined at contract level (as opposed to consensus level). Likely not part of v1.0.
```
Given that the current broken implementation has (parts of) its consensus in a contract it is safe to assume that whatever we build also will be a hybrid-solution.

hanssv commented on glossary.md:
```diff
@@ -0,0 +1,133 @@
+Contains one keyblock and possibly one microblock.
+
+### Keyblock
+A placeholder block indicating the leader that should produce a microblock.
```
Not sure about the placeholder comment. The keyblock contains all the state tree hashes etc so it is crucial for consensus. However, the separation into micro-block + key-block is entirely artificial and a legacy of building on top of a code base that uses BitcoinNG...

happi committed:
Add validator and adressing comments.

happi committed:
Add delegate.
